### PR TITLE
[ELY-1711] Tests suite split

### PR DESCRIPTION
**Pull request by skyllarr**
Tests from wildfly-elytron-tests that could be moved to its modules without introducing dependencies on other modules were moved. Remaining tests did either introduce some circular dependencies in maven or required additional dependencies.

### CODE REVIEW

File: wildfly-elytron-credential/pom.xml [hunk 1]
```diff
@@ -62,12 +62,36 @@
...
             <groupId>org.jboss.logging</groupId>
             <artifactId>jboss-logging-processor</artifactId>
             <scope>provided</scope>
-        </dependency>               
+        </dependency>
+
+        <!--Test scope-->
+        <dependency>
+            <groupId>org.wildfly.security</groupId>
+            <artifactId>wildfly-elytron-digest</artifactId>
```
Comment (darranl) [inline wildfly-elytron-credential/pom.xml:75]: Overall everything on this PR looks really good, it looks like most of the moved tests have moved cleanly without internal dependencies - what was the need for this dependency?  It looks like this module will now depend on digest where it did not previously?

**Comment by darranl** [comment]
FYI my comment may not stop us merging. but is a point we may want to revisit.

File: wildfly-elytron-credential/pom.xml (hunk 1 above)
Comment (skyllarr) [reply wildfly-elytron-credential/pom.xml:75]: If some tests did not move cleanly I left it in wildfly-elytron-tests. For this specific one, after addding this dependency all the tests passed. Without it the single test https://github.com/Skyllarr/wildfly-elytron/blob/tests-splitting/wildfly-elytron-credential/src/test/java/org/wildfly/security/password/impl/DigestPasswordTest.java#L94

will throw

```
testSHA512_256(org.wildfly.security.password.impl.DigestPasswordTest)  Time elapsed: 0.055 sec  <<< ERROR!
java.security.spec.InvalidKeySpecException: ELY08013: No such MessageDigest algorithm for "digest-sha-512-256"
	at org.wildfly.security.password.impl.DigestPasswordImpl.<init>(DigestPasswordImpl.java:82)
	at org.wildfly.security.password.impl.DigestPasswordImpl.<init>(DigestPasswordImpl.java:72)
	at org.wildfly.security.password.impl.PasswordFactorySpiImpl.engineGeneratePassword(PasswordFactorySpiImpl.java:261)
	at org.wildfly.security.password.PasswordFactory.generatePassword(PasswordFactory.java:151)
	at org.wildfly.security.password.impl.DigestPasswordTest.performTest(DigestPasswordTest.java:106)
	at org.wildfly.security.password.impl.DigestPasswordTest.testSHA512_256(DigestPasswordTest.java:96)
```

so I made an exception and added dependency here.

**Comment by darranl** [comment]
Just in case we want to revisit in the future I have added this issue https://issues.jboss.org/browse/ELY-1726

**Pull request merged**